src/sgx/signing/signing_untrusted.c

```diff
@@ -0,0 +1,132 @@
+/*
+ * This file contains Solana's SGX enclave code for signing data.
+ */
+
+#include <stdbool.h>
+#include <string.h>
+#include <time.h>
+
+#include "ed25519.h"
+#include "sgx_urts.h"
+#include "signing_public.h"
+#include "signing_u.h"
+
+static bool initialized = false;
+static bool use_enclave = false;
+static sgx_enclave_id_t eid;
+static uint8_t public_key[32], private_key[64];
+
+/* This function generates a random buffer of size length */
+static void untrusted_read_rand(unsigned char* val, size_t length) {
+  for (int i = 0; i < length; i++) {
+    val[i] = 0;
+    int num = rand();
+    while (num != 0) {
+      val[i] ^= num & 0xff;
+      num >>= 8;
+    }
+  }
+}
+
+/* This function creates a private/public keypair in the untrusted space.
+   It also initializes the untrusted random number generator using current
+   time. It's used only if the platform does not support SGX.
+*/
+static sgx_status_t init_untrusted(uint32_t keylen, uint8_t* pubkey) {
+  if (keylen < sizeof(public_key)) {
+    return SGX_ERROR_INVALID_PARAMETER;
+  }
+
+  time_t t = time(NULL);
+  struct tm* lt = (struct tm*)localtime((const time_t*)&t);
+  unsigned int rand_seed =
+      lt->tm_sec | lt->tm_min << 8 | lt->tm_hour << 16 | lt->tm_mday << 24;
+  srand(rand_seed);
+
+  uint8_t seed[32];
```

Comment:
why a seed of 32?  let's justify the magic number

Reply:
We'll probably have to use a const. It's there because ed25519 library expects these sizes, but does not have a typedef (neither it checks or returns an error).